fog/view/server/src/sharding_strategy.rs

```diff
@@ -140,4 +167,80 @@ mod epoch_sharding_strategy_tests {
 
         assert!(!should_process_block)
     }
+
+    #[test]
+    fn is_ready_to_serve_tx_outs_first_epoch_processed_block_count_is_zero_less_than_50_percent_processed_returns_true(
+    ) {
+        // The first epoch has a start block == 0.
+        const START_BLOCK: BlockIndex = 0;
+        const END_BLOCK_NON_INCLUSIVE: BlockIndex = 100;
+        let epoch_block_range = BlockRange::new(START_BLOCK, END_BLOCK_NON_INCLUSIVE);
+        let epoch_sharding_strategy = EpochShardingStrategy::new(epoch_block_range);
+
+        let is_ready_to_serve_txouts = epoch_sharding_strategy.is_ready_to_serve_tx_outs(0.into());
+
+        assert!(is_ready_to_serve_txouts)
+    }
+
+    #[test]
+    fn is_ready_to_serve_tx_outs_first_epoch_processed_block_count_is_zero_over_50_percent_processed_returns_true(
+    ) {
+        // The first epoch has a start block == 0.
+        const START_BLOCK: BlockIndex = 0;
+        const END_BLOCK_NON_INCLUSIVE: BlockIndex = 100;
+        let epoch_block_range = BlockRange::new(START_BLOCK, END_BLOCK_NON_INCLUSIVE);
+        let epoch_sharding_strategy = EpochShardingStrategy::new(epoch_block_range);
+
+        let is_ready_to_serve_txouts = epoch_sharding_strategy.is_ready_to_serve_tx_outs(70.into());
+
+        assert!(is_ready_to_serve_txouts)
+    }
+
+    #[test]
+    fn is_ready_to_serve_tx_outs_not_first_epoch_processed_block_count_less_than_50_percent_of_epoch_range_length_returns_false(
+    ) {
+        const START_BLOCK: BlockIndex = 100;
+        const END_BLOCK_NON_INCLUSIVE: BlockIndex = 110;
```

Comment:
Since this is integer division for halving suggest using an odd number
```suggestion
        const END_BLOCK_NON_INCLUSIVE: BlockIndex = 111;
```

Reply:
Done